### PR TITLE
fix_: add missing message verification flag when creating an account

### PR DESCRIPTION
**Pull request by richard-ramos**
cc: @cammellos and @ilmotta in case it's necessary to do something in mobile

### CODE REVIEW

**Review by plopezlpz — APPROVED** [review]

**Review by chaitanyaprem — APPROVED** [review]
LGTM

But defaults will be set from desktop/mobile is it?

**Review by kaichaosun — APPROVED** [review]

File: api/defaults.go
Comment (ilmotta) [inline api/defaults.go:308]: @richard-ramos, is there a recommendation that we always enable this flag on mobile? Would you say there's a scenario where we don't want this enabled on mobile?

**Review by ilmotta — APPROVED** [review]

File: api/defaults.go
Comment (richard-ramos) [reply api/defaults.go:308]: I imagine it should follow the same criteria as `WakuV2EnableStoreConfirmationForMessagesSent`

Comment (ilmotta) [reply api/defaults.go:308]: Okay 👍🏼 In which case we will have this setting disabled because we set `WakuV2EnableStoreConfirmationForMessagesSent` to false on profile creation from mobile. Thus the PR should require no changes from mobile. Thanks

Comment (chaitanyaprem) [reply api/defaults.go:308]: we can keep this enabled for mobile by default. The reason `WakuV2EnableStoreConfirmationForMessagesSent` was disabled is as it was giving bad UX to users where messages were sent but still showing as not sent due to storeNode performance issues. 

But this flag doesn't give any bad UX, so we can keep it enabled by default.

Comment (ilmotta) [reply api/defaults.go:308]: Sounds good @chaitanyaprem, thanks for the context.

Comment (richard-ramos) [reply api/defaults.go:308]: I wonder if we should enable it though, because this will do store queries constantly and might end up being expensive for mobile users.

Comment (richard-ramos) [reply api/defaults.go:308]: Perhaps this should be conditional similar to what is being done in this PR? https://github.com/status-im/status-go/pull/5511

Comment (ilmotta) [reply api/defaults.go:308]: Expensive from the battery perspective or from the mobile network? Or both @richard-ramos?

If it's related to the bandwidth cost, maybe this flag should be disabled dynamically when we detect the user switched to a mobile network. We could do that in the future. This is the direction we are going from this PR https://github.com/status-im/status-go/pull/5511

Comment (richard-ramos) [reply api/defaults.go:308]: Bandwidth cost. That flag what it will do is send a query every minute to retrieve the list of message hashes sent in the last minute.

Comment (richard-ramos) [reply api/defaults.go:308]: @ilmotta do you know if #5511 is going to land on 2.30? if so, i can try rebasing this PR over that one. Otherwise, i'll make it conditional on a separate PR

Comment (ilmotta) [reply api/defaults.go:308]: https://github.com/status-im/status-go/pull/5511 is hopefully going to land on 2.30 @richard-ramos. If you prefer to do directly in one status-go PR that seems fine as it's a very safe change, really up to you. On mobile side though, we would implement separately the dynamic usage of the flag because the PR https://github.com/status-im/status-mobile/pull/20730 is already rather large and about to be QAed (we have a separate issue for that https://github.com/status-im/status-mobile/issues/20879)

Since we are planning to cut the release branch tomorrow and are running very low on QA capacity due to other high-priority issues, the dynamic choice of the flag will require QA on mobile, and I'm afraid we can't put it on 2.30. Maybe a patch release like 2.30.1 for internal usage.

Comment (richard-ramos) [reply api/defaults.go:308]: In the meantime, to not hold this PR I added the following changes in https://github.com/status-im/status-go/pull/5570/commits/bd9f98f931443359a1cf5edea02c92554a8fb11a . If the connection being used is "expensive", skip the missing messages verification.  Seems to me like a good middle ground between not having the functionality enabled in mobile and consuming the users' data. Then once #5511 is merged, we can change the code to reuse the functionality to proceed with store queries depending on the users' choice.

@chaitanyaprem @ilmotta if you agree then we can proceed merge this PR with this change, otherwise i can roll it back.

Comment (ilmotta) [reply api/defaults.go:308]: Sounds great @richard-ramos 👍🏼

Comment (chaitanyaprem) [reply api/defaults.go:308]: I doubt the hash based store-query going to be as expensive as legacy-store. Because we only query hashes and if we miss any messages we fetch them which would improve message reliability. 

#5511 is expensive as it doesn't query hashes rather full messages.
I think we can keep this new flag enabled by default and provide a switch in the app in case it becomes expensive to disable it.

Comment (richard-ramos) [reply api/defaults.go:308]: alright. Worst case scenario we release a x.x.1 :)

**Pull request merged**
